crates/bevy_render/src/view/mod.rs

```diff
@@ -242,19 +242,47 @@ impl ViewTarget {
     ///
     /// A use case for this is to be able to prepare a bind group for all main textures
     /// ahead of time.
-    pub fn main_texture_other(&self) -> &TextureView {
+    pub fn main_texture_other(&self) -> &Texture {
         if self.main_texture.load(Ordering::SeqCst) == 0 {
             &self.main_textures.b
         } else {
             &self.main_textures.a
         }
     }
 
+    /// The "main" unsampled texture.
```

Comment:
IMO this should explain the difference with `main_texture`. It's surprising as a user to see two seemingly identical functions with the same documentation that returns two different types. What should I pick? I think a `view` is a subset of the texture right? Why would I use the `Texture` version over this one?

Reply:
I actually don't fully understand the distinction either, except that some APIs require `Texture` and others require `TextureView`. I guess it probably becomes more useful when you have an atlas or would like to "slice" portions of your texture for something, but I don't understand why calls (or most things?) can't take a `TextureView` directly instead of a `Texture`. `copy_texture_to_texture()` is especially weird given that it takes a texture _and_ a region, which could be better served by a `TextureView` instead?

Reply:
TextureView lets you do things like bind specific mip levels of a texture separately. They also correspond to different lower-level DX/VK/Metal objects, which I imagine is the real reason why WebGPU has both.

Reply:
Why not just return `CachedTexture` and let users pick what they need at the callsite?